website/www/site/content/en/documentation/ml/side-input-updates.md

```diff
@@ -0,0 +1,144 @@
+
+This example uses [`WatchFilePattern`](https://beam.apache.org/releases/pydoc/current/apache_beam.ml.inference.utils.html#apache_beam.ml.inference.utils.WatchFilePattern) as a side input. `WatchFilePattern` is used to watch for the file updates matching the `file_pattern`
+based on timestamps. It emits the latest [`ModelMetadata`](https://beam.apache.org/documentation/transforms/python/elementwise/runinference/), which is used in
+the RunInference `PTransform` to automatically update the ML model without stopping the Beam pipeline.
+
+## Set up the source
+
+To read the image names, use a Pub/Sub topic as the source.
+ * The Pub/Sub topic emits a `UTF-8` encoded model path that is used to read and preprocess images to run the inference.
+
+## Models for image segmentation
+
+For the purpose of this example, use TensorFlow models saved in [HDF5](https://www.tensorflow.org/tutorials/keras/save_and_load#hdf5_format) format.
+
+
+## Pre-process images for inference
+The Pub/Sub topic emits an image path. We need to read and preprocess the image to use it for RunInference. The `read_image` function is used to read the image for inference.
+
+```python
+import io
+from PIL import Image
+from apache_beam.io.filesystems import FileSystems
+import numpy
+import tensorflow as tf
+
+def read_image(image_file_name):
+  with FileSystems().open(image_file_name, 'r') as file:
+    data = Image.open(io.BytesIO(file.read())).convert('RGB')
+  img = data.resize((224, 224))
+  img = numpy.array(img) / 255.0
+  img_tensor = tf.cast(tf.convert_to_tensor(img[...]), dtype=tf.float32)
+  return img_tensor
+```
+
+Now, let's jump into the pipeline code.
+
+**Pipeline steps**:
+1. Get the image names from the Pub/Sub topic.
+2. Read and pre-process the images using the `read_image` function.
+3. Pass the images to the RunInference `PTransform`. RunInference takes `model_handler` and `model_metadata_pcoll` as input parameters.
+
+For the [`model_handler`](https://github.com/apache/beam/blob/07f52a478174f8733c7efedb7189955142faa5fa/sdks/python/apache_beam/ml/inference/base.py#L308), we use [TFModelHandlerTensor](https://github.com/apache/beam/blob/186973b110d82838fb8e5ba27f0225a67c336591/sdks/python/apache_beam/ml/inference/tensorflow_inference.py#L184).
+```python
+from apache_beam.ml.inference.tensorflow_inference import TFModelHandlerTensor
+# initialize TFModelHandlerTensor with a .h5 model saved in a directory accessible by the pipeline.
+tf_model_handler = TFModelHandlerTensor(model_uri='gs://<your-bucket>/<model_path.h5>')
+```
+
+The `model_metadata_pcoll` is a [side input](https://beam.apache.org/documentation/programming-guide/#side-inputs) `PCollection` to the RunInference `PTransform`. This side input is used to update the models in the `model_handler` without needing to stop the beam pipeline.
```

Comment:
@rszper this following section has a lot of code blocks which collectively make it kinda hard to read IMO (http://apache-beam-website-pull-requests.storage.googleapis.com/25947/documentation/ml/side-input-updates/index.html), but I don't know of a great way around it - any ideas on how we could make this section a little cleaner looking/easier to read? The content itself is correct